### ISSUE TITLE
Python treesitter parser should never be enabled

### ISSUE THREAD

**Issue by bavalpey** [NTBBloodbath/doom-nvim#223]
The Python Treesitter parser is in such a terrible state that it should not be supported by Doom (in the default config) until the parser is in much better shape.  Installing the parser causes python code editing to become more of a hassle than anything.

See

- https://github.com/nvim-treesitter/nvim-treesitter/issues/1136
- https://github.com/nvim-treesitter/nvim-treesitter/issues/1976

**Comment by NTBBloodbath** [comment]
Hi, thanks for pointing this out, being honest I didn't know this behavior with the Python treesitter parser.

What do you recommend to improve our Python support? :eyes:

**Comment by bavalpey** [comment]
Pyright, which is already present in doom via LspInstall, is perfect. I believe it it already does everything treesitter does (perhaps less performance, though).
In reality, this is a minor issue.  Really, I just think that enabling python support in the configuration files should default to pyright, and not enable python on treesitter.
FWIW, formatting and linting for python code is apparently very challenging for editors.  For instance, Vscode's python extension doesn't work either. Theres actually have an extension to fix the indentation, and it is both buggy and slow.

In other words, I don't think anything needs to be added, but rather taken away. This project has proven itself to be very polished, and it "just works." It's fantastic, and I love it.  However, the shipped configuration for python is anything but polished, and therefore shouldn't be present or available in the options :)

**Comment by mjlbach** [comment]
The functionality of pyright and treesitter is almost entirely orthogonal. Semantic tokens could hypothetically be used for tree sitter like queries(ish) (and perhaps highlighting) but it's not supported by pyright.

**Comment by connorgmeehan** [comment]
Closing due to inactivity.
